tgui/packages/tgui/components/TextArea.js

```diff
@@ -0,0 +1,184 @@
+      }
+      this.autoresize();
+    };
+    this.handleKeyDown = e => {
+      const { editing } = this.state;
+      const { onKeyDown } = this.props;
+      if (!editing) {
+        this.setEditing(true);
+      }
+      if (!dontUseTabForIndent) {
+        const keyCode = e.keyCode || e.which;
+        if (keyCode === 9) {
+          e.preventDefault();
+          const s = e.target.selectionStart;
+          e.target.value
+          = e.target.value.substring(0, e.target.selectionStart)
+            + "\t"
+            + e.target.value.substring(e.target.selectionEnd);
+          e.target.selectionEnd = s +1;
+        }
+      }
+
+      if (onKeyDown) {
+        onKeyDown(e, e.target.value);
+      }
+      this.autoresize();
+    };
+    this.handleFocus = e => {
+      const { editing } = this.state;
+      if (!editing) {
+        this.setEditing(true);
+      }
+    };
+    this.handleBlur = e => {
+      const { editing } = this.state;
+      const { onChange } = this.props;
+      if (editing) {
+        this.setEditing(false);
+        if (onChange) {
+          onChange(e, e.target.value);
+        }
+      }
+    };
+  }
+
+  componentDidMount() {
+    const nextValue = this.props.value;
+    const input = this.textareaRef.current;
+    if (input) {
+      input.value = toInputValue(nextValue);
+      this.autoresize();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { editing } = this.state;
+    const prevValue = prevProps.value;
+    const nextValue = this.props.value;
+    const input = this.textareaRef.current;
+    if (input && !editing && prevValue !== nextValue) {
+      input.value = toInputValue(nextValue);
+      this.autoresize();
+    }
+  }
+
+  setEditing(editing) {
+    this.setState({ editing });
+  }
+  getValue() {
+    return this.textareaRef.current && this.textareaRef.current.value;
+  }
+  render() {
+    const { props } = this;
+    // Input only props
+    const {
+      onChange,
+      onKeyDown,
+      onKeyPress,
+      onInput,
+      onFocus,
+      onBlur,
+      onEnter,
+      value,
+      placeholder,
+      ...boxProps
+    } = this.props;
+    // Box props
+    const {
+      className,
+      fluid,
+      ...rest
+    } = boxProps;
+    return (
+      <Box
+        className={classes([
+          'textarea-container',
+          'TextArea',
+          fluid && 'TextArea--fluid',
+          className,
+
+        ])}
+        {...rest}>
+
+        <textarea
+          value={value}
+          ref={this.textareaRef}
+          className="Textarea_textarea"
+          placeholder={placeholder}
+          onChange={this.handleOnChange}
+          onKeyDown={this.handleKeyDown}
+          onKeyPress={this.handleKeyPress}
+          onInput={this.handleOnInput}
+          onFocus={this.handleFocus}
+          onBlur={this.handleBlur}
+        />
+        <div className="TextArea__baseline">
+          .
+        </div>
+        <div ref={this.fillerRef} />
```

Comment:
Have you tried not using the filler div, but instead stretch textarea using:

```
position: absolute;
top: 0;
left: 0;
bottom: 0;
right: 0;
```

and this on `TextArea` class:

```
position: relative;
```